get_ac_audio.sh

```diff
@@ -20,32 +28,64 @@ fi
 
 unzip -o -d $TMP $ID.zip
 
-if ls $TMP/cameraVoip*.flv > /dev/null 2>&1; then
-    VOIP=($(ls $TMP/cameraVoip*.flv | sort --version-sort -f))
-elif ls $TMP/ftvoice*.flv > /dev/null 2>&1; then
-    VOIP=($(ls $TMP/ftvoice*.flv | sort --version-sort -f))
-elif ls $TMP/ftstage*.flv > /dev/null 2>&1; then
-    VOIP=($(ls $TMP/ftstage*.flv | sort --version-sort -f))
+if ls $TMP/*.mp3 > /dev/null 2>&1; then
+   VOIP=($(ls $TMP/*.mp3 | sort --version-sort -f))
+   echo 'At least 1 MP3 audio file is already available. Even if there are several - grabbing the first one to use as recording audio'
+   mv ${VOIP[0]} $OUTPUT_FILE
+   exit 0;
+else
+    ITEMS=(cameraVoip*.flv cameraVoip*.mp4 ftvoice*.flv ftstage*.flv)
+    for i in "${!ITEMS[@]}"; do
+        if ls $TMP/${ITEMS[$i]} > /dev/null 2>&1; then
+            LIST=($(ls $TMP/${ITEMS[$i]} | sort --version-sort -f))
+        else
+            echo "NOTICE: No items of type ${ITEMS[$i]} found. Skipping"
+            continue
+        fi
+        for j in "${!LIST[@]}"; do
+            if $FFPROBE_BIN -v error -show_entries stream=codec_type ${LIST[$j]} | grep -m1 -q audio; then
+                VOIP=($(ls $TMP/${ITEMS[$i]} | sort --version-sort -f))
+                break
+            fi
+        done
+    done
 fi
 
 if [ -z "$VOIP" ]; then
-    echo "$ID.zip does contains neither cameraVoip*.flv nor ftvoice*.flv files. Exiting:("
+    echo "ERR: $ID.zip contains no cameraVoip*.flv/cameraVoip*.mp4/ftvoice*.flv/ftstage*.flv files. Exiting:("
     exit 2
 fi
 
+rm -f $ID.list
 for i in "${!VOIP[@]}"; do
-    if ffprobe -v error -show_entries stream=codec_type ${VOIP[$i]} | grep -m1 -q audio; then
-        FILENAME=$(printf "%0*d" 4 $i)
-        ffmpeg -nostdin -i "${VOIP[$i]}" -y $ID/$FILENAME.mp3
+    if $FFPROBE_BIN -v error -show_entries stream=codec_type ${VOIP[$i]} | grep -m1 -q audio; then
+        FILE=${VOIP[$i]}
+        EXTENSION=${FILE#*.}
+        FILENAME=`basename "${FILE%%.*}" .$EXTENSION`
+
+        if [ $EXTENSION != 'mp4' ]; then
+            $FFMPEG_BIN -nostdin -i "${VOIP[$i]}" -y $ID/$FILENAME.mp3
+            EXTENSION=mp3
+        else
+            cp ${VOIP[$i]} $ID/
+        fi
+        echo "$OUTDIR/$ID/$FILENAME.$EXTENSION" >> $ID.list;
     fi
 done
-rm -f $ID.list
-for f in $ID/*.mp3; do echo "file '$PWD/$f'" >> $ID.list; done
+
+if [ -s $ID.list ]; then
+    echo "ERR: No audio files found. Exiting."
+    exit 2
```

Comment:
Hi @hilak,

Best exit with a different RC here since you already use `2` in line 56.
All the other modifications following the code review look good so we can merge when you're ready.